lib/error_catalog.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2018-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <map>
+#include <string>
+#include "error_catalog.h"
+
+// -------- Errors -------------
+const int ErrorType::LEGACY_ERROR      =   0;
```

Comment:
Why can't these be just an enum?
Do we care about their concrete values outside of the compiler?

Reply:
Per the class comment:
```
/// enumerate supported errors
/// Backends should extend this class with additional errors in the range 500-999 and
/// warnings in the range 1500-2141.
/// It is a class and not an enum class because in C++11 you can't extend an enum class
class ErrorType {
```